### PR TITLE
feat(api-v2): Add support for text file upload (DSP-44)

### PR DESCRIPTION
**Pull request by benjamingeer**
https://dasch.myjetbrains.com/youtrack/issue/DSP-44

- [x] Use MIME types in `application.conf` to check that the user has uploaded the correct file type for the value type they want to create.
- [x] Add workaround for https://dasch.myjetbrains.com/youtrack/issue/DSP-711.
- [x] Add tests.
- [x] Skip `ClientApiRouteE2ESpec` until we redesign it (https://dasch.myjetbrains.com/youtrack/issue/DSP-724).
- [x] Add docs.

### CODE REVIEW

**Comment by benjamingeer** [comment]
Skipping `ClientApiRouteE2ESpec` because it keeps failing with this error, despite everything we've tried: 

```
Failed to get test data: {"knora-api:error":"akka.pattern.AskTimeoutException: Ask timed out on
[Actor[akka://E2ETest/user/applicationManager#-1049986081]] after [5000 ms]. Message of type
[org.knora.webapi.messages.admin.responder.usersmessages.UserGetADM] was sent by
[Actor[akka://E2ETest/user/applicationManager#-1049986081]].
```

**Comment by subotic** [comment]
@benjamingeer @SepidehAlassi  could you please make sure to fix this test either in this PR or as the first next thing? If you decide to ignore it in this PR, please open an issue on YouTrack. This test is also failing on `develop`.

**Comment by benjamingeer** [comment]
> @benjamingeer @SepidehAlassi  could you please make sure to fix this test either in this PR or as the first next thing? If you decide to ignore it in this PR, please open an issue on YouTrack. This test is also failing on `develop`.

We're planning to redesign it in the next sprint, using your idea of collecting the test data as a side effect of running the E2E tests.

**Comment by subotic** [comment]
cool, great 👍

File: webapi/src/main/scala/org/knora/webapi/store/iiif/SipiConnector.scala [hunk 1]
```diff
@@ -232,20 +231,70 @@ class SipiConnector extends Actor with ActorLogging {
...
+        val knoraInfoUrl = getFileMetadataRequestV2.fileUrl + "/knora.json"
+        val sipiRequest = new HttpGet(knoraInfoUrl)
 
         for {
-            responseStr <- doSipiRequest(request)
-        } yield responseStr.parseJson.convertTo[GetFileMetadataResponseV2]
+            sipiResponseStr <- doSipiRequest(sipiRequest)
+            sipiResponse: SipiKnoraJsonResponse = sipiResponseStr.parseJson.convertTo[SipiKnoraJsonResponse]
+
+            // Workaround for https://dasch.myjetbrains.com/youtrack/issue/DSP-711
```
Comment (SepidehAlassi) [inline webapi/src/main/scala/org/knora/webapi/store/iiif/SipiConnector.scala:281]: @benjamingeer it seems like Lukas has already fixed the issue of DSP-711, see [his response](https://dasch.myjetbrains.com/youtrack/issue/DSP-711#focus=Comments-4-1112.0-0) should this workaround be removed in this PR or later?

**Review by SepidehAlassi — CHANGES_REQUESTED** [review]
Thanks for this, please see my comment regarding work around for DSP-711

**Review by SepidehAlassi — APPROVED** [review]
@benjamingeer thanks for your work. As agreed, please make a new PR with the new version of SIPI.

**Comment by subotic** [comment]
PR for updating Sipi: #1721

**Comment by benjamingeer** [comment]
@SepidehAlassi Thanks for reviewing!

**Pull request merged**